src/Data/Functor/Rep/Internal.hs

```diff
@@ -432,22 +447,92 @@ tabulateCotraverse1 =
 indexLogarithm :: f a -> Logarithm f -> a
 indexLogarithm = flip runLogarithm
 
--- | Derive 'cotraverse1' via an isomorphism
-cotraverse1Iso ::
-     (Representable g, Functor1 w)
-  => (forall x. f x -> g x)
-  -> (forall x. g x -> f x)
-  -> (w Identity -> a)
-  -> w f
-  -> f a
-cotraverse1Iso t frm f = frm . cotraverseMap1 f t
-
-gcotraverse1 ::
+cotraverseMap1Iso ::
+     (Representable f', Functor1 w)
+  => (forall x. f x -> f' x)
+  -> (f' a -> f a)
+  -> (w Identity -> a) -> (forall x. g x -> f x) -> w g -> f a
+cotraverseMap1Iso too frm f g = frm . cotraverseMap1 f (too . g)
+
+gcotraverseMap1 ::
      (Representable (Rep1 f), Functor1 w, Generic1 f)
-  => (w Identity -> a)
-  -> w f
-  -> f a
-gcotraverse1 = cotraverse1Iso from1 to1
+  => (w Identity -> a) -> (forall x. g x -> f x) -> w g -> f a
+gcotraverseMap1 = cotraverseMap1Iso from1 to1
+
+-- | A version of 'cotraverseMap1Iso' when the isomorphism is actually a
+-- coercion. This function is only exported from "Data.Functor.Rep" for GHC >=
+-- 8.6. A completely unsafe version is available in "Data.Functor.Rep.Internal"
+-- for earlier GHC versions.
+#if __GLASGOW_HASKELL__ >= 806
+cotraverseMap1Coerce ::
+     ( Representable f', Functor1 w
+     , forall x. Coercible (f' x) (f x) )
+  => (forall x. f' x `arr` f x)
+  -> (w Identity -> a) -> (forall x. g x -> f x) -> w g -> f a
```

Comment:
Hmmm.... Since we can't get `Compose` here for some reason, I wonder how bad it would be to use the hammer of `Coercible f f'`....

Reply:
Nope, that would lose more.